src/routes/policies.rs

```diff
@@ -41,15 +43,26 @@ pub async fn create_policy(
     _auth: ApiKey,
     policy: Json<schemas::Policy>,
     policy_store: &State<Box<dyn PolicyStore>>,
+    schema_store: &State<Box<dyn SchemaStore>>,
 ) -> Result<Json<schemas::Policy>, AgentError> {
     let policy = policy.into_inner();
-    let added_policy = policy_store.create_policy(policy.borrow()).await;
+    let schema = schema_store.get_cedar_schema().await;
+
+    let added_policy = policy_store.create_policy(policy.borrow(), schema).await;
     match added_policy {
         Ok(p) => Ok(Json::from(p)),
-        Err(_) => Err(AgentError::Duplicate {
-            id: policy.id,
-            object: "policy",
-        }),
+        Err(e) => {
+            if let Some(PolicyStoreError::PolicyInvalid(_, reason)) = e.downcast_ref::<PolicyStoreError>() {
```

Comment:
pretty sure this can be simplified using something similar to [this](https://doc.rust-lang.org/book/ch09-02-recoverable-errors-with-result.html#matching-on-different-errors)

Reply:
it isn't a blocking feedback and mostly semantic